src/test/rpc/AccountObjects_test.cpp

```diff
@@ -316,10 +316,225 @@ class AccountObjects_test : public beast::unit_test::suite
+            jvPayChan["PublicKey"] = strHex (gw.pk().slice ());
+            env (jvPayChan);
+            env.close();
+        }
+        {
+            // Find the payment channel.
+            Json::Value const resp = acct_objs (gw, jss::payment_channel);
+
+            BEAST_EXPECT (resp["result"]["account_objects"].isArray());
+            BEAST_EXPECT (resp["result"]["account_objects"].size() == 1);
+
+            auto const& payChan = resp["result"]["account_objects"][0u];
+            BEAST_EXPECT (payChan["Account"] == gw.human());
+            BEAST_EXPECT (payChan["Amount"].asInt() == 300'000'000);
+            BEAST_EXPECT (payChan["SettleDelay"].asInt() == 24 * 60 * 60);
+        }
+        // Make gw multisigning by adding a signerList.
+        env (signers (gw, 6, { { alice, 7} }));
+        env.close();
+        {
+            // Find the signer list.
+            Json::Value const resp = acct_objs (gw, jss::signer_list);
+
+            BEAST_EXPECT (resp["result"]["account_objects"].isArray());
+            BEAST_EXPECT (resp["result"]["account_objects"].size() == 1);
+
+            auto const& signerList = resp["result"]["account_objects"][0u];
+            BEAST_EXPECT (signerList["SignerQuorum"] == 6);
+            auto const& entry = signerList["SignerEntries"][0u]["SignerEntry"];
+            BEAST_EXPECT (entry["Account"] == alice.human());
+            BEAST_EXPECT (entry["SignerWeight"].asInt() == 7);
+        }
+        // Create a Ticket for gw.
+        env (ticket::create (gw, gw));
+        env.close();
+        {
+            // Find the ticket.
+            Json::Value const resp = acct_objs (gw, jss::ticket);
+
+            BEAST_EXPECT (resp["result"]["account_objects"].isArray());
+            BEAST_EXPECT (resp["result"]["account_objects"].size() == 1);
+
+            auto const& ticket = resp["result"]["account_objects"][0u];
+            BEAST_EXPECT (ticket["Account"] == gw.human());
+            BEAST_EXPECT (ticket["LedgerEntryType"] == "Ticket");
+            BEAST_EXPECT (ticket["Sequence"].asInt() == 8);
+        }
+        // Run up the number of directory entries so gw has two
+        // directory nodes.
+        for (int d = 1'000'032; d >= 1'000'000; d -= 1)
+        {
+            env (offer (gw, USD (1), drops (d)));
+            env.close();
+        }
+
+        // Make a lambda that easily identifies null account objects.
+        auto acct_objs_is_null =
+            [&env, &acct_objs] (Account const& acct, char const* type)
+        {
+            Json::Value const resp = acct_objs (acct, type);
+            return resp["result"]["account_objects"].isNull();
+        };
+
+        // Verify that the non-returning types still don't return anything.
+        // Note that, interestingly, they now return a null object rather
+        // than an empty array.  Not sure what triggers that...
+        BEAST_EXPECT (acct_objs_is_null (gw, jss::account));
```

Comment:
These are essentially jibberish, unrecognized types ?

Reply:
Nevermind, I see. I thought `amendments` and `hashes` was non sensical in the context of account_objects.  So `non-returning types`  means there was just no results?

Reply:
I guess it's this line here: 
https://github.com/ripple/rippled/blob/develop/src/ripple/rpc/impl/RPCHelpers.cpp#L100

It is *probably* the weird json library, which creates a null node, until you run append on it. 

This is probably the "fix":
```c++
 auto& jvObjects = jvResult[jss::account_objects] = Json::Value(Json::objectValue);
```